ebi_eva_common_pyutils/ncbi_utils.py

```diff
@@ -94,3 +94,23 @@ def retrieve_species_scientific_name_from_tax_id_ncbi(taxid):
     match = re.search('<ScientificName>(.+?)</ScientificName>', r.text, re.MULTILINE)
     if match:
         return match.group(1)
+
+
+def get_species_name_from_ncbi(assembly_acc):
+    # We first need to search for the species associated with the assembly
+    assembly_dicts = get_ncbi_assembly_dicts_from_term(assembly_acc)
+    taxid_and_assembly_name = set([
+        (assembly_dict.get('taxid'), assembly_dict.get('assemblyname'))
+        for assembly_dict in assembly_dicts
+        if assembly_dict.get('assemblyaccession') == assembly_acc or
+           assembly_dict.get('synonym', {}).get('genbank') == assembly_acc
+    ])
```

Comment:
Courtesy @apriltuesday for this tip!
```suggestion
taxid_and_assembly_name = {
        (assembly_dict.get('taxid'), assembly_dict.get('assemblyname'))
        for assembly_dict in assembly_dicts
        if assembly_dict.get('assemblyaccession') == assembly_acc or
           assembly_dict.get('synonym', {}).get('genbank') == assembly_acc
    }
```